resources/services/monitor/diagnostic_settings.go

```diff
@@ -217,14 +216,10 @@ func fetchMonitorDiagnosticSettings(ctx context.Context, meta schema.ClientMeta,
 	}
 
 	g, _ := errgroup.WithContext(ctx)
-	limiter := semaphore.NewWeighted(10)
+	g.SetLimit(10)
 	for _, i := range ids {
 		id := i
 		g.Go(func() error {
-			if err := limiter.Acquire(ctx, 1); err != nil {
```

Comment:
This has a minor issue. While it limits the `List` operations it still creates an number of go routines as the number of ids